doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/examples/ml-example.md

```diff
@@ -1,10 +1,253 @@
+
+Once you have set up your Kubernetes cluster, deploy the KubeRay operator:
+```shell
+kubectl create -k "github.com/ray-project/kuberay/ray-operator/config/default?ref=v0.3.0-rc.0"
+```
+
+## Deploying a Ray cluster
+
+Now we're ready to deploy the Ray cluster that will execute our workload.
+
+:::{tip}
+The Ray cluster we'll deploy is configured such that one Ray pod will be scheduled
+per 16-CPU Kubernetes node. The pattern of one Ray pod per Kubernetes node is encouraged, but not required.
+Broadly speaking, it is more efficient to use a few large Ray pods than many small ones.
+:::
+
+You may choose to deploy either a statically-sized Ray cluster or an autoscaling
+Ray cluster. Run one of the two commands below to deploy your Ray cluster.
+
+### Deploying a fixed-sized Ray cluster
+
+This option is most appropriate if you have set up a statically sized Kubernetes
+node pool or group.
+
+We recommend taking a look at the config file applied in the following command.
+```shell
+# Starting from the parent directory of cloned Ray master,
+pushd ray/doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/configs/
+kubectl apply -f xgboost-benchmark.yaml
+popd
+```
+
+A Ray head pod and 9 Ray worker pods will be created.
+
+### Deploying an autoscaling Ray cluster
+
+This option is most appropriate if you have set up an autoscaling Kubernetes
+node pool or group.
+
+We recommend taking a look at the config file applied in the following command.
+```shell
+# Starting from the parent directory of cloned Ray master,
+pushd ray/doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/configs/
+kubectl apply -f xgboost-benchmark-autoscaler.yaml
+popd
+```
+
+One Ray head pod will be created. Once the workload starts, the Ray autoscaler will trigger
+creation of Ray worker pods. Kubernetes autoscaling will then create nodes to place the Ray pods.
+
+
+## Running the workload
+
+To observe the startup progress of the Ray head pod, run the following command.
+```shell
+# If you're on MacOS, first `brew install watch`.
+watch -n 1 kubectl get pod
+```
+
+Once the Ray head pod enters `Running` state, we are ready to execute the XGBoost workload.
+We will use {ref}`Ray Job Submission<jobs-overview>` to kick off the workload.
+
+### Connect to the cluster.
+
+First, we connect to the Job server. Run the following blocking command
+in a separate shell.
+```shell
+kubectl port-forward service/raycluster-xgboost-benchmark-head-svc 8265:8265
+```
+
+### Submit the workload.
+
+We'll use the {ref}`Ray Job Python SDK<ray-job-sdk>` to submit the XGBoost workload.
+
+```{literalinclude} ../doc_code/xgboost_submit.py
+:language: python
+```
+
+To submit the workload, run the above Python script.
+The script is available in the Ray repository.
+
+```shell
+# From the parent directory of cloned Ray master.
+pushd ray/doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/doc_code/
+python xgboost_submit.py
```

Comment:
TODO in a follow-up: Provide a link to this script rather than referencing a Ray repo directory path.